lib/katello/tasks/pulp3_migration_stats.rake

```diff
@@ -3,17 +3,39 @@ namespace :katello do
   task :pulp3_migration_stats => [:environment] do
     User.current = User.anonymous_admin
 
-    migrated_rpms = ::Katello::Rpm.where('pulp_id LIKE ?', '%/pulp/api/v3/content/rpm/packages/%').
+    migrated_rpm_count = ::Katello::Rpm.where('pulp_id LIKE ?', '%/pulp/api/v3/content/rpm/packages/%').
       or(::Katello::Rpm.where.not(migrated_pulp3_href: nil)).count
-    migrated_errata = ::Katello::RepositoryErratum.where.not(erratum_pulp3_href: nil).count
-    migrated_repos = ::Katello::Repository.where.not(version_href: nil).count
-    migratable_repos = ::Katello::Repository.count - ::Katello::Repository.puppet_type.count -
+    migrated_erratum_count = ::Katello::RepositoryErratum.where.not(erratum_pulp3_href: nil).count
+    migrated_repo_count = ::Katello::Repository.where.not(version_href: nil).count
+    migratable_repo_count = ::Katello::Repository.count - ::Katello::Repository.puppet_type.count -
       ::Katello::Repository.ostree_type.count - ::Katello::Repository.deb_type.count
 
+    immediate_rpms = ::Katello::Repository.yum_type.collect do |repo|
+      repo.rpms unless repo.on_demand?
+    end
```

Comment:
I think you could turn this into a query that would be much faster.  Also i would imagine if you can stick to rpm ids, it'd be faster too

something like:
Katello::RepositoryRpm.where(:repository_id => Katello::Repository.yum_type).select(:id).uniq

Reply:
updated previous comment to fix .select()

Reply:
Yeah, I got a bit stuck on the on_demand check. I'll have to think about how to add that in.

Reply:
i think there's an 'on_demand' scope

Reply:
Also `.distinct` may be a bit faster if that's possible, since it will do the duplicate detection via SQL instead of Ruby.

Reply:
oh yeah, and since really all you care about is the count, you can drop the select:

Katello::RepositoryRpm.where(:repository_id => Katello::Repository.yum_type.on_demand).distinct.count